### ISSUE TITLE
On partitioned topics, DLQ producer create panics

### ISSUE THREAD

**Issue by maraiskruger1980** [apache/pulsar-client-go#568]
#### Expected behavior

When a message from a partitioned topic is put on DLQ it should work

#### Actual behavior

It panics with the following error:

```
panic: runtime error: invalid memory address or nil pointer dereference
19
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x5e4a14]
18
17
goroutine 86 [running]:
16
github.com/apache/pulsar-client-go/pulsar/internal.(*lookupService).GetPartitionedTopicMetadata(0x400021ce10, 0x4000042102, 0x32, 0x0, 0x0, 0x0)
15
	/go/pkg/mod/github.com/!marais/pulsar-client-go@v1.0.3/pulsar/internal/lookup_service.go:193 +0x1f4
14
github.com/apache/pulsar-client-go/pulsar.(*client).TopicPartitions(0x400021b3e0, 0x4000042102, 0x32, 0x4001a58a00, 0x400024daa8, 0x1c0e8, 0xb0, 0xffffbd2caec0)
13
	/go/pkg/mod/github.com/!marais/pulsar-client-go@v1.0.3/pulsar/client_impl.go:184 +0x6c
12
github.com/apache/pulsar-client-go/pulsar.(*producer).internalCreatePartitionsProducers(0x40003c2140, 0x0, 0x0)
11
	/go/pkg/mod/github.com/!marais/pulsar-client-go@v1.0.3/pulsar/producer_impl.go:161 +0x54
10
github.com/apache/pulsar-client-go/pulsar.newProducer(0x400021b3e0, 0x40004b4e40, 0x400008dc90, 0x1c120, 0x400008dd08)
9
	/go/pkg/mod/github.com/!marais/pulsar-client-go@v1.0.3/pulsar/producer_impl.go:123 +0x214
8
github.com/apache/pulsar-client-go/pulsar.(*client).CreateProducer(0x400021b3e0, 0x4000042102, 0x32, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
7
	/go/pkg/mod/github.com/!marais/pulsar-client-go@v1.0.3/pulsar/client_impl.go:153 +0x70
6
github.com/apache/pulsar-client-go/pulsar.(*dlqRouter).getProducer(0x400021ce60, 0x40006f1260, 0x1)
5
	/go/pkg/mod/github.com/!marais/pulsar-client-go@v1.0.3/pulsar/dlq_router.go:139 +0xc0
4
github.com/apache/pulsar-client-go/pulsar.(*dlqRouter).run(0x400021ce60)
3
	/go/pkg/mod/github.com/!marais/pulsar-client-go@v1.0.3/pulsar/dlq_router.go:92 +0x1ac
2
created by github.com/apache/pulsar-client-go/pulsar.newDlqRouter
1
```

#### Steps to reproduce

Create a consumer that NACKS all messages on a partitioned topic. When it tries to create the DLQ producer, it panics

#### System configuration
**Pulsar version**: 2.6

**Comment by cckellogg** [comment]
What version of the client are you using?

**Comment by maraiskruger1980** [comment]
0.5.0

**Comment by cckellogg** [comment]
Will you try with the latest release 0.6.0 and see if the issue has been fixed?

**Comment by maraiskruger1980** [comment]
@cckellogg This seems to be fixed now un 0.6.0 Can't seem to replicate the issue
